mono/metadata/appdomain.c

```diff
@@ -2467,9 +2467,14 @@ ves_icall_System_Reflection_Assembly_InternalLoad (MonoStringHandle name_handle,
 	gboolean parsed;
 	char *name;
 
+	MonoAssembly *requesting_assembly = mono_runtime_get_caller_from_stack_mark (stack_mark);
 	MonoAssemblyLoadContext *alc = (MonoAssemblyLoadContext *)load_Context;
+
+	if (!alc)
+		alc = mono_assembly_get_alc (requesting_assembly);
 	if (!alc)
-		alc = mono_domain_default_alc (mono_domain_get ());
+		g_assert_not_reached ();
```

Comment:
this is `g_assert (alc)`.  The same as line 2479.  Maybe just move that up to 2475?

Reply:
Ah whoops, yep.